cmd/demo/node_setup.go

```diff
@@ -132,6 +143,13 @@ func (n *node) setupContracts() (err error) {
 			return err
 		}
 		assAddr = ewallet.AsEthAddr(tmpAsset)
+
+		ctx, cancel := context.WithTimeout(context.Background(), config.Chain.TxTimeout)
+		defer cancel()
+		err = echannel.ValidateAssetHolderETH(ctx, n.cb, assAddr, adjAddr)
+		if err != nil {
+			return err
+		}
```

Comment:
```suggestion
		if err := echannel.ValidateAssetHolderETH(ctx, n.cb, assAddr, adjAddr); err != nil {
			return errors.WithMessage(err, "validating asset holder contract")
		}
```
note that the message should describe the "action", usually in gerund form, because it chains nicely when displaying the error like "Error doing foo: accessing bar: validating contract: opening bla: connection refused".

Reply:
suggested changes have been adopted to the codebase